src/vss_tools/vspec/vssexporters/vss2apigear.py

```diff
@@ -0,0 +1,506 @@
+    return property
+
+
+def export_node(node: VSSNode, module: ApiGearModule, interface: ApiGearInterface | None = None):
+    log.debug(f"Node {node.get_fqn()}")
+    data = node.get_vss_data()
+    if isinstance(node.data, VSSDataBranch):
+        new_interface_name = node_name(node)
+        new_interface = ApiGearInterface(new_interface_name)
+        module.interfaces.append(new_interface)
+
+        for child in node.children:
+            export_node(child, module, new_interface)
+    else:
+        allowed = getattr(data, "allowed")
+        if allowed:
+            enum_name = f"{node_name(node)}_Value"
+            enum = ApiGearEnumeration(enum_name)
+            log.debug(f"Creating a new enum: {enum_name}")
+
+            for element in allowed:
+                element_str = str(element)
+                element_dec = hex_to_dec(element_str)
+                variant = get_allowed_enum_literal(element_str)
+                log.debug(f"  New variant: {variant}")
+                datatype = getattr(data, "datatype")
+                if datatype == Datatypes.FLOAT[0]:
+                    enum.variants.append((variant, None))
+                else:
+                    enum.variants.append((variant, element_dec))
+
+            property = ApiGearProperty(ApiGearType(enum_name))
+            if data.description != "":
+                property.description = data.description
+            if interface is not None:
+                interface.properties[node.name] = property
+                module.enumerations.append(enum)
+        else:
+            datatype = getattr(data, "datatype")
+            if not datatype:
+                log.warning(f"Node {node.get_fqn()} does not have a datatype")
+                return
+            prop = generate_property(node, datatype)
+            if prop is not None and interface is not None:
+                interface.properties[node.name] = prop
+
+
+def export_data_type_node(node: VSSNode | None, module: ApiGearModule, structure: ApiGearStructure | None = None):
+    """This method is used to traverse through the root VSS Data Types node to build
+       -> ApiGear equivalent string buffer and to serialize it accordingly into a module
+    """
+    if node is None:
+        return
+
+    data = node.get_vss_data()
+    log.debug(f"Node [{data.type}] {node.get_fqn()}")
+
+    if isinstance(node.data, VSSDataBranch):
+        for child in node.children:
+            export_data_type_node(child, module)
+    elif isinstance(node.data, VSSDataStruct):
+        struct_name = node_name(node)
+        struct = ApiGearStructure(struct_name)
+        log.debug(f"Creating a new structure {struct_name}")
+
+        for child in node.children:
+            export_data_type_node(child, module, struct)
+
+        module.structures.append(struct)
+    else:
+        datatype = getattr(data, "datatype")
+        if not datatype:
+            log.warning(f"Node {node.get_fqn()} does not have a datatype")
```

Comment:
We can derive whether a node has a datatype from its instance type. Pydantic ensures that a node that should have a datatype also actually has one

Reply:
I'm not sure if I followed you correctly, but I've removed this check and replaced it with datatype instance check: `isinstance(node.data, VSSDataDatatype)`.